### PR TITLE
Start a chapter on Documentation

### PR DESCRIPTION
**Pull request by mkuzak**
This PR addresses issue #879 about adding the chapter on documentation

### Summary

Fixes #<NUM>

*Lorem ipsum dolor sit amet, consectetur adipiscing.*

### List of changes proposed in this PR (pull-request)

* *Lorem ipsum dolor sit amet, consectetur adipiscing.*
* *Lorem ipsum dolor sit amet, consectetur adipiscing.*

### What should a reviewer concentrate their feedback on?

- [ ] *Lorem ipsum dolor sit amet, consectetur adipiscing.*
- [ ] Everything looks ok?

### Acknowledging contributors

- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/alan-turing-institute/the-turing-way/blob/master/README.md#contributors) in the README file:

### CODE REVIEW

**Comment by malvikasharan** [comment]
@mkuzak I have added label for this content/chapter to go to both collaboration and communication books. What do you think?

**Comment by sofievdbos** [comment]
Two possible GitHub options: [_WIP_]
* Continuous Documentation: https://readthedocs.org/
* Publish as website with _docs_ or _gh-pages_ as source: https://help.github.com/en/enterprise/2.13/user/articles/configuring-a-publishing-source-for-github-pages

File: book/content/documentation/documenation.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+## About
+Both the project and the source code should be documentated in order for others to understand what the project is about, how they can use it and how they can contribute to it.
+
+### Markdown
+[Markdown](https://en.wikipedia.org/wiki/Markdown) is a lightweight markup language with plain-text-formatting syntax. Its design allows it to be converted to many output formats, but the original tool by the same name only supports HTML. Markdown is often used to format readme files, for writing messages in online discussion forums, and to create rich text using a plain text editor.
+
+GitHub and other project and code hosting services automatically render markdown files for humans to read. 
+
+You can learn about markdown syntax and how to master this language in an excellent GitHug [Mastering markdown Guide](https://guides.github.com/features/mastering-markdown/). Try it out with an [online editor](https://markdown-it.github.io)
```
Comment (sverhoeven) [inline book/content/documentation/documenation.md:11]: Markdown syntax is a bit ambiguous, better to use [CommonMark](https://commonmark.org/). Can we replace this section with CommonMark sites like https://commonmark.org/help/ and  https://spec.commonmark.org/dingus/

File: book/content/documentation/documenation.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+### Code of Conduct (link to collaboration chapter)
+
+### Software Citation (expand section in recognition chapter)
+
+## Code Documentation
+
+### code comments
+
+### API documentation
```
Comment (sverhoeven) [inline book/content/documentation/documenation.md:38]: For a library the API documentation is the same as the code documention aka description of classes/functions/constants etc.. 

For a web service there are standards that like [OpenAPI](https://www.openapis.org/) or [Graphql](https://graphql.org/).

File: book/content/documentation/documenation.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### usage documentation
```
Comment (sverhoeven) [inline book/content/documentation/documenation.md:40]: For a command line interface you could make a [man page](https://en.wikipedia.org/wiki/Man_page) with examples.

**Comment by malvikasharan** [comment]
@jmfernandez Can you please review this?

File: book/content/documentation/documenation.md (hunk 1 above)
Comment (jmfernandez) [inline book/content/documentation/documenation.md:11]: There is a typo, `GitHug` should be `GitHub`

File: book/content/documentation/documenation.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+GitHub and other project and code hosting services automatically render markdown files for humans to read. 
+
+You can learn about markdown syntax and how to master this language in an excellent GitHug [Mastering markdown Guide](https://guides.github.com/features/mastering-markdown/). Try it out with an [online editor](https://markdown-it.github.io)
+
+## Project Documentation
+Describe your project for others to understand what it is about, what is the audience, how others can contribute to it. The [Mozilla Open Leadership Training Series](https://mozilla.github.io/open-leadership-training-series/) [Opening Your Project Chapter](https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/) is a good starting point to learn about how to efectively communicate information abou your project.
```
Comment (jmfernandez) [inline book/content/documentation/documenation.md:14]: Italics, bold text or simialr should be used to indicate that `Opening Your Project Chapter` is a different link from _Mozilla OpenLeadership Training Series_

File: book/content/documentation/documenation.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Documentation
+
+## About
+Both the project and the source code should be documentated in order for others to understand what the project is about, how they can use it and how they can contribute to it.
```
Comment (jmfernandez) [inline book/content/documentation/documenation.md:4]: Typo: `documentated` should be `documented`

File: book/content/documentation/documenation.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+### CHANGELOG
+https://keepachangelog.com/en/1.0.0/
+
+### LICENSE (link to license section)
+
+### Code of Conduct (link to collaboration chapter)
+
+### Software Citation (expand section in recognition chapter)
+
+## Code Documentation
```
Comment (jmfernandez) [inline book/content/documentation/documenation.md:34]: A very nice explanation, which could be either referenced or reused in some way (if its license allows it)

https://medium.com/@andrewgoldis/how-to-document-source-code-responsibly-2b2f303aa525

Example of Rosseta Commons community

https://www.rosettacommons.org/docs/latest/development_documentation/Coding-Conventions

**Review by jmfernandez — COMMENTED** [review]
I have left several comments, some about typos, other ones adding external sources which could be used

File: book/content/documentation/documenation.md (hunk 1 above)
Comment (c-martinez) [reply book/content/documentation/documenation.md]: This part of the document has been updated.

File: book/content/documentation/documenation.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+### CONTRIBUTING (link to collaboration chapter)
+[Contributor guidelines](https://mozilla.github.io/open-leadership-training-series/articles/building-communities-of-contributors/write-contributor-guidelines/), by convension are described in the file called CONTRIBUTING. They are the nuts and bolts instructions on how to contribute to your project. Learn more on how to write effective guidelines in Getting Contributors section.
+
+### ROADMAP
+[Example of ROADMAP file](https://docs.readthedocs.io/en/stable/roadmap.html#roadmap) from Read The Docs project.
+
+### CHANGELOG
+https://keepachangelog.com/en/1.0.0/
+
+### LICENSE (link to license section)
```
[the remark this answers is not in the document]
Comment (c-martinez) [reply book/content/documentation/documenation.md:26]: Probably this should be in the license section.

File: book/content/documentation/documenation.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
+# Documentation
+
+## About
+Both the project and the source code should be documentated in order for others to understand what the project is about, how they can use it and how they can contribute to it.
+
+### Markdown
+[Markdown](https://en.wikipedia.org/wiki/Markdown) is a lightweight markup language with plain-text-formatting syntax. Its design allows it to be converted to many output formats, but the original tool by the same name only supports HTML. Markdown is often used to format readme files, for writing messages in online discussion forums, and to create rich text using a plain text editor.
+
+GitHub and other project and code hosting services automatically render markdown files for humans to read.
```
[the remark this answers is not in the document]
Comment (c-martinez) [reply book/content/documentation/documenation.md]: Good point -- adding a link in the project documentation part.

File: book/content/documentation/documenation.md (hunk 6 above)
Comment (c-martinez) [reply book/content/documentation/documenation.md:32]: I've included the links, but am not sure about the license.

**Review by jmfernandez — COMMENTED** [review]
I have just realized the filename has a typo, as it should be `documentation.md` instead of _documenation.md_ . But renaming it may imply changes in other files referencing this.

**Comment by c-martinez** [comment]
This PR does not touch the table of contents (yet) so renaming should not be a problem.

**Comment by c-martinez** [comment]
This was a proposal, but it is by now out of date. Closing this issue for now, and maybe we should revive it in the future.

**Pull request closed without merging**